File: qif4j/src/main/java/com/github/squirrelgrip/qif/write/AbstractQifRecord.java
package com.github.squirrelgrip.qif.write;

public abstract class AbstractQifRecord implements QifRecord {
	public String asFormattedRecord(QifHeaderEnum previousHeader) {
		StringBuilder sb = new StringBuilder();
		if (previousHeader == null) {
			sb.append(getHeader().getText()).append(Utils.LINE_SEPARATOR);
		}
		sb.append(formatTransaction()).append(Utils.LINE_SEPARATOR);

		sb.append(Utils.END_OF_RECORD).append(Utils.LINE_SEPARATOR);
		return sb.toString();
	}

	public abstract QifHeaderEnum getHeader();

	public abstract String formatTransaction();
}

File: qif4k/src/main/kotlin/com/github/squirrelgrip/qif4k/write/AbstractQifRecord.kt
package com.github.squirrelgrip.qif4k.write

abstract class AbstractQifRecord : QifRecord {
    override fun asFormattedRecord(previousHeader: QifHeaderEnum?): String {
        val sb = StringBuilder()
        if (previousHeader == null) {
            sb.append(header!!.text).append(Utils.LINE_SEPARATOR)
        }
        sb.append(formatTransaction()).append(Utils.LINE_SEPARATOR)
        sb.append(Utils.END_OF_RECORD).append(Utils.LINE_SEPARATOR)
        return sb.toString()
    }

    abstract override val header: QifHeaderEnum?
    abstract fun formatTransaction(): String?
}